c/uca/sifter/unisifeggs.cpp

```diff
@@ -0,0 +1,433 @@
+#include <algorithm>
+#include <array>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <ranges>
+#include <string>
+#include <string_view>
+#include <vector>
+
+class CheckFailure {
+ public:
+  ~CheckFailure() {
+    std::terminate();
+  }
+  void operator&(std::ostream&) {}
+};
+
+#define CHECK(condition) \
+  (condition)            \
+      ? (void)0          \
+      : CheckFailure() & std::cerr << ("Check failed: " #condition "\n")
+
+class CodePointRange {
+ public:
+  constexpr CodePointRange(char32_t front, char32_t back)
+      : front_(front), back_(back) {
+    if (empty()) {
+      front_ = 1;
+      back_ = 0;
+    }
+  }
+
+  constexpr bool empty() const {
+    return front_ > back_;
+  }
+
+  constexpr bool contains(char32_t c) const {
+    return c >= front_ && c <= back_;
+  }
+
+  constexpr CodePointRange intersection(CodePointRange const& other) const {
+    return CodePointRange(std::max(front_, other.front_),
+                          std::min(back_, other.back_));
+  }
+
+  constexpr char32_t front() const {
```

Comment:
... getStart(), getEnd(), maybe even getLimit() if there are call sites that add 1 ...

Reply:
The name should be `front` to match conventions for C++ sequence containers (`front()` for the first element). See https://en.cppreference.com/w/cpp/named_req/SequenceContainer.

Reply:
Gosh these names are bad. push_back() makes some sense, where "back" indicates *where* you push, not *what* you push. front() and back() for the first/last elements are just sucky names...